bin/tests-github-actions.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -eu
+
+function get_seeded_random() {
+  openssl enc -aes-256-ctr -pass pass:"vimeo/psalm" -nosalt </dev/zero 2>/dev/null
+}
+
+function run {
+  local -r chunk_count="$1"
+  local -r chunk_number="$2"
+  local -r parallel_processes="$3"
+
+  local -r phpunit_cmd='
+echo "::group::{}";
+vendor/bin/phpunit --log-junit build/phpunit/logs/{_}.xml --colors=always {};
+exit_code=$?;
+echo ::endgroup::;
+if [[ "$exit_code" -ne 0 ]]; then
+    echo "::error::{}";
+fi;
+exit "$exit_code"'
+
+  mkdir -p build/parallel/ build/phpunit/logs/
+
+  find tests -name '*Test.php' | shuf --random-source=<(get_seeded_random) > build/tests_all
```

Comment:
This doesn't sound right to me: this line shuffles the test based on a seed that is unique for each spawned `jobs.tests`, because `get_seeded_random` gets called indeed inside each of them.

So every `job` shuffles them differently and this lead to some test being run twice or worse, not being run at all.

The `random-source` should be generated only once in the `chuck-matrix` job.

/cc @weirdan @danog

Reply:
the random source is consistent.
you can verfy this youself by running this multiple times
`find tests -name '*Test.php' | shuf --random-source=<( openssl enc -aes-256-ctr -pass pass:"vimeo/psalm" -nosalt </dev/zero 2>/dev/null)`

Reply:
source
https://www.gnu.org/software/coreutils/manual/html_node/Random-sources.html#Random-sources

Reply:
Yeah, I confirm I always saw error in jobs to be consistent from one run to another

Reply:
> the random source is consistent.

This is good news indeed.

I wonder why `shuf` is needed in the first place then :thinking: 
Maybe you know you have some long-running tests and other quick tests, and you want to spread them evenly across the `jobs`? It would be good to write a comment about the why, just to clear the things out